javascriptv3/example_code/cloudwatch-events/tests/cloudwatch-events.integration.test.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { describe, it, expect, beforeAll, afterAll } from "vitest";
+import {
+  CloudWatchLogsClient,
+  CreateLogGroupCommand,
+  DeleteLogGroupCommand,
+  DescribeLogGroupsCommand,
+} from "@aws-sdk/client-cloudwatch-logs";
+import {
+  DeleteRuleCommand,
+  CloudWatchEventsClient,
+  RemoveTargetsCommand,
+} from "@aws-sdk/client-cloudwatch-events";
+import { DEFAULT_REGION } from "../../libs/utils/util-aws-sdk.js";
+
+/** Create a log group.
+ * @param {string} logGroupName - The name of the log group.
+ */
+const createLogGroup = async (logGroupName) => {
+  const cwlClient = new CloudWatchLogsClient({ region: DEFAULT_REGION });
+  const command = new CreateLogGroupCommand({
+    logGroupName: logGroupName,
+  });
+
+  await cwlClient.send(command);
+
+  // Describe the log group to get the ARN.
+  const describeLogGroupsCommand = new DescribeLogGroupsCommand({
+    logGroupNamePrefix: logGroupName,
+  });
+  const data = await cwlClient.send(describeLogGroupsCommand);
+  return data.logGroups[0].arn;
+};
+
+/**
+ * Delete a rule.
+ * @param {string} ruleName - The name of the rule.
+ */
+const deleteRule = async (ruleName) => {
+  const cweClient = new CloudWatchEventsClient({ region: DEFAULT_REGION });
+  const command = new DeleteRuleCommand({
+    Name: ruleName,
+  });
+  await cweClient.send(command);
+};
+
+/**
+ * Remove a target from a rule.
+ * @param {string} ruleName - The name of the rule.
+ * @param {string[]} targetIds- The IDs of the targets to remove.
+ */
+const removeTargets = async (ruleName, targetIds) => {
+  const cweClient = new CloudWatchEventsClient({ region: DEFAULT_REGION });
+  const command = new RemoveTargetsCommand({
+    Rule: ruleName,
+    Ids: targetIds,
+  });
+  await cweClient.send(command);
+};
+
+/**
+ * Delete a log group.
+ * @param {string} logGroupName - The name of the log group.
+ */
+const deleteLogGroup = async (logGroupName) => {
+  const cwlClient = new CloudWatchLogsClient({ region: DEFAULT_REGION });
+  const command = new DeleteLogGroupCommand({
+    logGroupName: logGroupName,
+  });
+  await cwlClient.send(command);
+};
+
+describe("CloudWatch Events integration test", () => {
+  const logGroupName = "test-log-group";
+  const ruleName = "test-rule";
+  const rulePattern = JSON.stringify({
+    source: ["my.app"],
+    "detail-type": ["My Custom Event"],
+  });
+  const targetId = "test-target";
+  let targetArn = "";
```

Comment:
(Opt) You might be interested in [cleanState](https://github.com/jefri/jiffies/blob/9437f6f60707f8a2f69ac4e2c3177c9cdf0c957e/src/scope/state.ts#L1) ([usage](https://github.com/nand2tetris/web-ide/blob/a6eef1c63bc15324746a7a7a62baf1c5f1f246de/simulator/src/output.test.ts#L25-L37))

Reply:
That's very clean looking, but took me a bit to understand. Is this a common convention you've seen?
I like the mental load for reading test cases to be minimal.

Reply:
It prevents leaking state kept in scope between tests.